packages/configurationmap/components/ConfigurationMapHeading.tsx

```diff
@@ -0,0 +1,44 @@
+import * as React from "react";
+import { cx } from "emotion";
+import { padding, border } from "../../shared/styles/styleUtils";
+import { configurationMapHeading } from "../style";
+
+interface ConfigurationMapHeadingProps {
+  /**
+   * What to render as the heading content
+   */
+  children: React.ReactNode;
+  /**
+   * Priority of the heading. Numbers map to <h1> through <h6>
+   */
+  headingLevel?: 1 | 2 | 3 | 4 | 5 | 6;
```

Comment:
> 1. Do we really need 6 level deep headings
A `HashMap` with nested objects increases it's heading level (up to `6`) with each level of nesting

> 2. How this is implemented we could have a headingLevel 7 in the default value since that one does not get type checked if we want to have a default value we need a implementation like this:

I don't think I'm following what you're saying here. Maybe we could hop on a Zoom and talk more?

> 3. I would prefer if we would expose ConfigurationMapHeading1... ConfigurationMapHeading6 as different components instead of 1 flexible component.

Why do you prefer that?

Reply:
Let's talk about this offline it is also not blocking.